### PR TITLE
Neobauer

### LINKED ISSUE #3: reading input files function

### PR DESCRIPTION
**Pull request by NeoBauer**
resolve #3

### CODE REVIEW

File: bowling/directoryContent.hpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+class directoryContent {
+public:
+  directoryContent();
+  directoryContent(char dir[]);
+  std::vector<fs::path> getFilesInDirectory() const;
+  std::map<std::string, std::vector<std::string>> getFileNameAndContentMap() const;
+private:
+  fs::path directory_;
+  std::vector<fs::path> filesInDirectory;
```
Comment (J-Cieplinski) [inline bowling/directoryContent.hpp]:
```suggestion
  std::vector<fs::path> filesInDirectory_;
```

File: bowling/directoryContent.hpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  void searchFilesInDirectory();
+  void sortFilesInVector();
+  void readFiles();
+  std::map<std::string, std::vector<std::string>> fileNameAndContentMap;
```
Comment (J-Cieplinski) [inline bowling/directoryContent.hpp]:
```suggestion
  std::map<std::string, std::vector<std::string>> fileNameAndContentMap_;
```

File: bowling/directoryContent.hpp [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#include <algorithm>
+#include <filesystem>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+namespace fs = std::filesystem;
+
+class directoryContent {
```
Comment (J-Cieplinski) [inline bowling/directoryContent.hpp]:
```suggestion
class DirectoryContent {
```

File: bowling/directoryContent.cpp [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+};
+
+void directoryContent::searchFilesInDirectory() {
+  if (fs::exists(directory_) && fs::is_directory(directory_)) {
+    for (auto const &entry : fs::directory_iterator(directory_)) {
+      auto pathName = entry.path();
+      if (fs::is_regular_file(entry.status()))
+        filesInDirectory.emplace_back(entry.path());
+      else if (fs::is_directory(entry.status())) {
+        std::cout << "You have some unknow folder named:" << pathName.filename()
```
Comment (J-Cieplinski) [inline bowling/directoryContent.cpp]:
```suggestion
        std::cout << "You have some unknown folder named:" << pathName.filename()
```

File: bowling/tests/test.cpp [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+          std::cout << "Expected Vector: \n";
+          for (auto &string : v) {
+            std::cout << string << '\n';
+          }
+          std::cout << "Original Map: " << originalMap.find(k)->first << '\n';
+          std::cout << "Original Vector: \n";
+          for (auto &string : originalMap.find(k)->second) {
+            std::cout << string << '\n';
+          }
+        } */
```
Comment (J-Cieplinski) [inline bowling/tests/test.cpp]: Why was this left in?

**Review by J-Cieplinski — CHANGES_REQUESTED** [review]

File: bowling/directoryContent.cpp (hunk 4 above)
Comment (J-Cieplinski) [inline bowling/directoryContent.cpp]: is it even needed? Wouldn't it be better to just ignore if its a directory and not check it?

File: bowling/directoryContent.cpp [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  sortFilesInVector();
+  readFiles();
+};
+
+void directoryContent::searchFilesInDirectory() {
+  if (fs::exists(directory_) && fs::is_directory(directory_)) {
+    for (auto const &entry : fs::directory_iterator(directory_)) {
+      auto pathName = entry.path();
+      if (fs::is_regular_file(entry.status()))
+        filesInDirectory.emplace_back(entry.path());
```
Comment (J-Cieplinski) [inline bowling/directoryContent.cpp]: Could this be replaced with std::for_each using https://en.cppreference.com/w/cpp/filesystem/directory_iterator/begin ?

File: .vscode/settings.json [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        "numeric": "cpp",
+        "regex": "cpp",
+        "set": "cpp",
+        "fstream": "cpp",
+        "future": "cpp",
+        "mutex": "cpp",
+        "thread": "cpp",
+        "variant": "cpp"
+    }
+}
```
Comment (KGebski0036) [inline .vscode/settings.json:70]: add new line

File: bowling/DirectoryContent.cpp [hunk 8]
```diff
@@ -0,0 +1,55 @@
+#include "DirectoryContent.hpp"
+#include <fstream>
+namespace fs = std::filesystem;
```
Comment (KGebski0036) [inline bowling/DirectoryContent.cpp:3]: one space line and here shouldn't be using?

File: bowling/FlagTests.cpp [hunk 9]
```diff
@@ -0,0 +1,5 @@
+#include "FlagTests.hpp"
+
+void printOutHelp(std::ostream& output){
+    output << "HelpMeIamLost";
+}
```
Comment (ksoczowka) [inline bowling/FlagTests.cpp:5]: add new line

File: bowling/FlagTests.hpp [hunk 10]
```diff
@@ -0,0 +1,4 @@
+#pragma once
+#include <iostream>
+
+void printOutHelp(std::ostream& output);
```
Comment (ksoczowka) [inline bowling/FlagTests.hpp:4]: add new line

File: bowling/tests/tests.cpp [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    int flagsCount = 2;
+    char *flagsContent[] = {
+        "ExecutedFileName",
+        "h",
+    };
+    WHEN("PassMainArguments") {
+      SearchHelpFlag(flagsCount, flagsContent);
+      THEN("CheckIfFlagWasFound") {}
+    }
+  } */
```
Comment (ksoczowka) [inline bowling/tests/tests.cpp]: apply or delete commented code

**Review by ksoczowka — DISMISSED** [review]

File: bowling/DirectoryContent.cpp [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+  sortFilesInVector();
+  readFiles();
+};
+
+void DirectoryContent::searchFilesInDirectory() {
+  if (fs::exists(directory_) && fs::is_directory(directory_)) {
+    for (auto const &entry : fs::directory_iterator(directory_)) {
+      auto pathName = entry.path();
+      if (fs::is_regular_file(entry.status()))
+        filesInDirectory_.emplace_back(entry.path());
```
Comment (KGebski0036) [inline bowling/DirectoryContent.cpp:18]: add {}

File: bowling/DirectoryContent.hpp [hunk 13]
```diff
@@ -0,0 +1,23 @@
+#include <algorithm>
```
Comment (KGebski0036) [inline bowling/DirectoryContent.hpp:1]: why u add this in this file?

File: bowling/DirectoryContent.hpp [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+#include <filesystem>
+#include <iostream>
```
Comment (KGebski0036) [inline bowling/DirectoryContent.hpp:3]: why u add this in this file?

File: bowling/DirectoryContent.hpp [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+#include <map>
+#include <string>
+#include <vector>
```
Comment (KGebski0036) [inline bowling/DirectoryContent.hpp:6]: #pragma once?

File: bowling/main.cpp [hunk 16]
```diff
@@ -0,0 +1,13 @@
+#include "DirectoryContent.hpp"
+#include <filesystem>
+#include <iostream>
```
Comment (KGebski0036) [inline bowling/main.cpp:3]: why iostream?

File: bowling/main.cpp [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace fs = std::filesystem;
+
+int main() {
+char dir[] = "../scores";
+DirectoryContent ScoresFolder(dir);
```
Comment (KGebski0036) [inline bowling/main.cpp:9]: indentation

File: bowling/tests/tests.cpp (hunk 11 above)
Comment (KGebski0036) [inline bowling/tests/tests.cpp]: UGLY COMMENT xP

**Review by KGebski0036 — CHANGES_REQUESTED** [review]

File: bowling/DirectoryContent.hpp (hunk 14 above)
Comment (NeoBauer) [reply bowling/DirectoryContent.hpp:3]: > why u add this in this file?

Filesystem is for path object - to iterate through folder

**Review by J-Cieplinski — APPROVED** [review]

**Review by KGebski0036 — APPROVED** [review]

**Pull request merged**